runtime/src/accounts_index.rs

```diff
@@ -1235,30 +1235,28 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
     /// Remove the slot when the storage for the slot is freed
     /// Accounts no longer reference this slot.
     pub fn clean_dead_slot(&self, slot: Slot) -> Option<AccountsIndexRootsStats> {
-        if self.is_root(slot) {
-            let (roots_len, uncleaned_roots_len, previous_uncleaned_roots_len, roots_range) = {
-                let mut w_roots_tracker = self.roots_tracker.write().unwrap();
-                w_roots_tracker.roots.remove(&slot);
-                w_roots_tracker.uncleaned_roots.remove(&slot);
-                w_roots_tracker.previous_uncleaned_roots.remove(&slot);
-                (
-                    w_roots_tracker.roots.len(),
-                    w_roots_tracker.uncleaned_roots.len(),
-                    w_roots_tracker.previous_uncleaned_roots.len(),
-                    w_roots_tracker.roots.range_width(),
-                )
-            };
-            Some(AccountsIndexRootsStats {
-                roots_len,
-                uncleaned_roots_len,
-                previous_uncleaned_roots_len,
-                roots_range,
-                rooted_cleaned_count: 0,
-                unrooted_cleaned_count: 0,
-            })
-        } else {
-            None
-        }
+        let (roots_len, uncleaned_roots_len, previous_uncleaned_roots_len, roots_range) = {
+            let mut w_roots_tracker = self.roots_tracker.write().unwrap();
+            if !w_roots_tracker.roots.remove(&slot) {
+                return None;
+            }
+            w_roots_tracker.uncleaned_roots.remove(&slot);
```

Comment:
should we go ahead and remove this slot in case it does exist in these?

Reply:
I think that integrity check is not a bad idea, to collect metrics to see if this is actually happening

Reply:
will add a metric separately